### ISSUE TITLE
Remove unnecessary mocks from tests

### ISSUE THREAD

**Issue by maraoz** [OpenZeppelin/zeppelin-solidity#259]
Many tests use mocks where there's no need for one. Most tests could use the contracts directly. Go through all tests and evaluate if a mock is necessary (for example, when the tested contract is abstract), and remove it where it isn't.

If you do this: THANKS!

**Comment by rstormsf** [comment]
@maraoz  Could you help me out understand the issue, so I can help

**Comment by maraoz** [comment]
When the contract is abstract, using a "Mock" (the name mock is not even correct) makes it possible to test it, because abstract tests cannot be instantiated. 
But when a contract is not abstract, you could just use the contract in tests, no need to write a "mock".

**Comment by maraoz** [comment]
Mocks are also useful when testing a contract which exposes function modifiers you want to test

**Comment by rudygodoy** [comment]
I've been looking at this, at least 6 of them. Most of them cannot be easily de-mocked because they lack of external interfaces, so mocha cannot interact with the contract for most of the test cases. `LimitBalance.sol` and `DayLimit.sol` looked good candidates for de-mocking. What would you advise?

I've refactored one test, not sure if PR is in order with regards to this issue though. https://github.com/rudygodoy/zeppelin-solidity/commit/549590d10515038c879e251293929c68115d9f6d

**Comment by frangio** [comment]
@rudygodoy Actually those two can't be de-mocked either because they provide modifiers...

We had the impression that some of them were unnecessary, but might have been wrong. It would be great if you could go through the rest and see if any are indeed unnecessary and remove them. We'd appreciate it!

**Comment by rudygodoy** [comment]
I went through all of them. Couldn't find a case where de-mocking was possible. I've encountered and fixed some typos in comments and did the `beforeEach(...);` refactor when it was possible. All test pass and I've validated that the introduction of such changes doesn't change the purpose/nature of each test case.

Individual commits can be found here: https://github.com/rudygodoy/zeppelin-solidity/commits/master Let me know if a PR is in order so I can do it for those small changes.
